toolbox/utilities/db_helper.py

```diff
@@ -17,27 +21,43 @@ def execute_sql_query(sql_query, db_host=Config.DB_HOST, extra_options=""):
 
 
 @contextlib.contextmanager
-def connect_to_read_replica():
+def connect_to_read_replica_pool():
+    conn_pool = None
+
     try:
-        conn = psycopg2.connect(f"dbname='{Config.DB_NAME}' user='{Config.DB_USERNAME}' host='{Config.DB_HOST}' "
-                                f"password='{Config.DB_PASSWORD}' port='{Config.DB_PORT}'{Config.DB_USESSL}")
-        yield conn
+        conn_pool = pool.SimpleConnectionPool(1, 2, user=Config.DB_USERNAME,
+                                              password=Config.DB_PASSWORD,
+                                              host=Config.DB_HOST,
+                                              port=Config.DB_PORT,
+                                              database=Config.DB_NAME)
+        yield conn_pool
     finally:
-        conn.close()
+        if conn_pool:
+            conn_pool.closeall()
 
 
-def execute_in_connection(*args, conn=None):
-    cursor = conn.cursor()
-    cursor.execute(*args)
-    return cursor.fetchall()
+@retry(wait_exponential_multiplier=1000, wait_exponential_max=10000)
```

Comment:
I believe with these settings it'll retry forever, should we have a max attempts?

Reply:
yes, great idea